### PR TITLE
**Fix** UI-41 Modal rules update

### PR DESCRIPTION
**Pull request by mpotomin**
# ToDo: 
- [x] rebase on master before the merge

# Summary
1. Makes `fullsize` modal shrink to 400px height min if content is small enough

Small content:
Screenshot 2019-06-12 at 12 58 13

Big content is still limited by the size of the page, utilises scroll:
Screenshot 2019-06-12 at 12 58 06

2. Fixes the background of the underlying card getting narrower than the modal itself if modal's content is too wide
Screenshot 2019-06-12 at 12 57 13

# Related issue
https://contiamo.atlassian.net/browse/UI-41

**Issue about rewriting the controlled page modal:**
https://github.com/contiamo/operational-ui/issues/1025

# To be tested

Me
- [ ] No error or warning in the console on `localhost:6060`

Tester 1
- [ ] Things look good on the demo.
- [ ] No other modal breaks

Tester 2
- [ ] Things look good on the demo.
- [ ] No other modal breaks

### CODE REVIEW

File: src/Internals/ControlledModal.tsx [hunk 1]
```diff
@@ -36,14 +36,16 @@ const fromTop = (fullSize: boolean) => {
 
 export const ModalContainer = styled(Card)<{ fullSize: boolean; type?: ControlledModalProps["type"] }>(
   ({ theme, fullSize, type }) => ({
+    label: "ModalContainer",
     top: theme.space.element,
     left: fullSize ? theme.space.element : "50%",
-    height: fullSize ? "100%" : "fit-content",
+    maxHeight: `calc(100% - ${theme.space.element * 3}px)`,
```
Comment (stereobooster) [inline src/Internals/ControlledModal.tsx]: why exactly 3? maybe small comment on why

**Comment by stereobooster** [comment]
@mpotomin is there a way to test it in our deploy preview? I can't find it

**Comment by TejasQ** [comment]
@stereobooster go to `Layout` examples and scroll down

**Comment by mpotomin** [comment]
Added the preview example at the very bottom of the Page component

**Pull request merged**